src/ClientData/TimerData.cpp

```diff
@@ -63,6 +65,43 @@ std::vector<const orbit_client_protos::TimerInfo*> TimerData::GetTimers(uint64_t
   return timers;
 }
 
+std::vector<const orbit_client_protos::TimerInfo*> TimerData::GetTimersAtDepthDiscretized(
+    uint32_t depth, uint32_t resolution, uint64_t start_ns, uint64_t end_ns) const {
+  ORBIT_SCOPE_WITH_COLOR("GetTimersAtDepthDiscretized", kOrbitColorBlueGrey);
+  absl::MutexLock lock(&mutex_);
+  // The query is for the interval [start_ns, end_ns], but it's easier to work with the close-open
+  // interval [start_ns, end_ns+1). We have to be careful with overflowing.
+  end_ns = std::max(end_ns, end_ns + 1);
+
+  if (timers_.find(depth) == timers_.end()) return {};
+
+  std::vector<const orbit_client_protos::TimerInfo*> discretized_timers;
+  uint64_t next_pixel_start_ns = start_ns;
+
+  // We are iterating through all blocks, but we don't have other option if we use TimerChain. A
```

Comment:
I didn't get the comment. But I think with the early return, it is not needed.

Reply:
I'm changing the implementation as we talked offline.

Reply:
The comment has been modified.